pkg/services/ngalert/schedule/schedule.go

```diff
@@ -487,10 +509,11 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 				}
 			}()
 		case <-grafanaCtx.Done():
-			// clean up the state only if the reason for stopping the evaluation loop is that the rule was deleted
-			if errors.Is(grafanaCtx.Err(), errRuleDeleted) {
-				clearState()
-			}
+			// We need to use context.Background() here because grafanaCtx is being canceled, and we need a valid context
+			// in clearState function to do DB operations.
```

Comment:
Since this was changed from `context.Background()` to `context.WithTimeout()`:

```suggestion
			// We need to use a new context here because grafanaCtx is being canceled, and we need a valid context
			// in clearState function to do DB operations.
```